src/cache.rs

```diff
@@ -0,0 +1,394 @@
+                    k.to_string()
+                ))
+                .await?;
+        }
+
+        let mut connection = self.pool.get().await.unwrap();
+        let response: ClientResult<CachedSkytableValue> = connection
+            .query_parse(&query!(
+                format!("select * from {} where ckey = ?", self.build_model()).as_str(),
+                k.to_string()
+            ))
+            .await;
+
+        match response {
+            Ok(response) => Ok(serde_json::from_str(response.cvalue.as_str())?),
+            Err(error) => match error {
+                skytable::error::Error::ServerError(111) => Ok(None),
+                error => Err(SkytableCacheError::from(error)),
+            },
+        }
+    }
+
+    #[instrument(skip_all)]
+    async fn cache_set(&self, k: K, v: V) -> std::result::Result<Option<V>, Self::Error> {
+        let value = CachedSkytableValue::new(
+            k.to_string(),
+            serde_json::to_string(&v)?,
+            Utc::now().timestamp() + self.seconds as i64,
+        );
+        self.pool
+            .get()
+            .await
+            .unwrap()
+            .query(&query!(
+                format!("insert into {}(?, ?, ?)", self.build_model()).as_str(),
+                value
+            ))
+            .await?;
+
+        Ok(Some(v))
+    }
+
+    async fn cache_remove(&self, k: &K) -> std::result::Result<Option<V>, Self::Error> {
+        let response: ClientResult<CachedSkytableValue> = self
+            .pool
+            .get()
+            .await
+            .unwrap()
+            .query_parse(&query!(
+                format!("select * from {} where key = ?", self.build_model()).as_str(),
+                k.to_string()
+            ))
+            .await;
+
+        self.pool
+            .get()
+            .await
+            .unwrap()
+            .query(&query!(
+                format!("delete from {} where key = ?", self.build_model()).as_str(),
+                k.to_string()
+            ))
+            .await?;
+
+        match response {
+            Ok(response) => Ok(serde_json::from_str(response.cvalue.as_str())?),
+            Err(error) => match error {
+                skytable::error::Error::ServerError(111) => Ok(None),
+                error => Err(SkytableCacheError::from(error)),
+            },
+        }
+    }
+
+    fn cache_set_refresh(&mut self, refresh: bool) -> bool {
+        let old = self.refresh;
+        self.refresh = refresh;
+        old
+    }
+}
+
+// TODO:find a way to implement optional tls here
+#[io_cached(
+    map_error = r##"|e| FeedbackFusionError::ConfigurationError(format!("{:?}", e))"##,
+    ty = "SkytableCache<skytable::pool::ConnectionMgrTcp, String, Option<Prompt>>",
+    create = r##" {
+        skytable_configuration!()
+    } "##,
+    convert = r#"{ format!("prompt-{}", id) }"#
+)]
+pub async fn fetch_prompt(connection: &DatabaseConnection, id: &str) -> crate::Result<Option<Prompt>> {
```

Comment:
somehow add a prefix to this functions and try to support tls variants via proc macro